### PR TITLE
Added unit tests to RequiresAccountConfirmationAttribute, ApiAuthorizeAttribute

### PR DESCRIPTION
**Pull request by TerraVenil**

### CODE REVIEW

**Comment by maartenba** [comment]
These unit tests are definitely welcome, thanks for that!

Can you remove the C#6 features used? (nameof) - our build is still on an older version for now, we'll do the C# 6 upgrade later.

File: src/NuGetGallery/Filters/ApiAuthorizeAttribute.cs [hunk 1]
```diff
@@ -15,7 +13,6 @@ protected override void HandleUnauthorizedRequest(AuthorizationContext filterCon
         {
             var owinContext = filterContext.HttpContext.GetOwinContext();
             owinContext.Authentication.Challenge(AuthenticationTypes.ApiKey);
-            owinContext.Response.StatusCode = 401;
```
Comment (maartenba) [inline src/NuGetGallery/Filters/ApiAuthorizeAttribute.cs]: Keep this line.

File: src/NuGetGallery/Filters/RequiresAccountConfirmationAttribute.cs [hunk 2]
```diff
@@ -22,7 +20,7 @@ public override void OnActionExecuting(ActionExecutingContext filterContext)
         {
             if (filterContext == null)
             {
-                throw new ArgumentNullException("filterContext");
+                throw new ArgumentNullException(nameof(filterContext));
```
Comment (maartenba) [inline src/NuGetGallery/Filters/RequiresAccountConfirmationAttribute.cs]: Don't use C# features - our build does not support it yet.

File: src/NuGetGallery/Filters/ApiAuthorizeAttribute.cs (hunk 1 above)
Comment (TerraVenil) [inline src/NuGetGallery/Filters/ApiAuthorizeAttribute.cs]: Can you explain why this line is nessesary? As noted previously the same logic already exists inside OWIN middleware when you are call method Challenge. Or this is some specific case?
Line 173 AuthenticationManager.cs
_context.Response.StatusCode = 401;

Comment (maartenba) [inline src/NuGetGallery/Filters/ApiAuthorizeAttribute.cs]: Looking at commit history I can't shake off the feeling we need this one.

**Comment by TerraVenil** [comment]
Created a new PR to dev branch.

**Comment by maartenba** [comment]
Closed in favor of https://github.com/NuGet/NuGetGallery/pull/2682

**Pull request closed without merging**